src/backend/win_cid/file_dialog/dialog_ffi.rs

```diff
@@ -88,11 +86,12 @@ impl IDialog {
 
             for f in filters.iter() {
                 let name: Vec<u16> = OsStr::new(&f.name).encode_wide().chain(once(0)).collect();
-                let mut ext_string = String::new();
-
-                for e in f.extensions.iter() {
-                    ext_string += &format!("*.{};", e);
-                }
+                let ext_string = f
+                    .extensions
+                    .iter()
+                    .map(|item| format!("*.{}", item))
+                    .collect::<Vec<_>>()
+                    .join(";");
```

Comment:
So... the last item should not have `;` at the end. Did I understand that right?

Reply:
Ahh, yeah.  I was gonna make a separate PR fixing this!  VSCode for example only put a `;` if there are multiple items in the list.

Reply:
I forgot to make a separate branch for this fix, so it got picked up here.

Reply:
No problem, commit name is descriptive enough. I just wanted to make sure that I understand this fix correctly.